packages/glimmer-reference/lib/iterable.ts

```diff
@@ -138,22 +143,27 @@ export class ReferenceIterator {
     this.artifacts = artifacts;
   }
 
-  next(): IterationItem<PathReference<Opaque>> {
+  getPosition(): UpdatableReference< number | FIXME<'user str to InternedString'>> {
+    return new UpdatableReference(this.iterator.getPosition());
+  }
+
+  next(): ListItem {
     let { artifacts } = this;
 
     let iterator = (this.iterator = this.iterator || artifacts.iterate());
 
     let item = iterator.next();
+    let position = iterator.getPosition();
 
     if (!item) return null;
 
-    return artifacts.append(item);
+    return artifacts.append(item, position);
```

Comment:
I'm not sure why position is sometimes part of item and sometimes not.  Why isn't it like value where the list item is updated?